Linkage error

Hello, I tried to compile the library on Windows 10 and 11, but I got linkage errors.
I tested on both lastest release and lastest commit.

Compiler information:
* Visual Studio 17.3.4
* MSVC 14.33.31629

Linkage with sentencepiece_train library

```
trainer_interface.obj : error LNK2001: unresolved external symbol "private: static class google::protobuf::internal::La
zyString const sentencepiece::TrainerSpec::_i_give_permission_to_break_this_code_default_unk_piece_" (?_i_give_permissi
on_to_break_this_code_default_unk_piece_@TrainerSpec@sentencepiece@@0VLazyString@internal@protobuf@google@@B) [C:\Users
\leres\Documents\GitHub\sentencepiece\src\sentencepiece_train.vcxproj]
sentencepiece_trainer.obj : error LNK2001: unresolved external symbol "private: static class google::protobuf::internal
::LazyString const sentencepiece::TrainerSpec::_i_give_permission_to_break_this_code_default_unk_piece_" (?_i_give_perm
ission_to_break_this_code_default_unk_piece_@TrainerSpec@sentencepiece@@0VLazyString@internal@protobuf@google@@B) [C:\U
sers\leres\Documents\GitHub\sentencepiece\src\sentencepiece_train.vcxproj]
trainer_interface.obj : error LNK2001: unresolved external symbol "private: static class google::protobuf::internal::La
zyString const sentencepiece::TrainerSpec::_i_give_permission_to_break_this_code_default_bos_piece_" (?_i_give_permissi
on_to_break_this_code_default_bos_piece_@TrainerSpec@sentencepiece@@0VLazyString@internal@protobuf@google@@B) [C:\Users
\leres\Documents\GitHub\sentencepiece\src\sentencepiece_train.vcxproj]
sentencepiece_trainer.obj : error LNK2001: unresolved external symbol "private: static class google::protobuf::internal
::LazyString const sentencepiece::TrainerSpec::_i_give_permission_to_break_this_code_default_bos_piece_" (?_i_give_perm
ission_to_break_this_code_default_bos_piece_@TrainerSpec@sentencepiece@@0VLazyString@internal@protobuf@google@@B) [C:\U
sers\leres\Documents\GitHub\sentencepiece\src\sentencepiece_train.vcxproj]
trainer_interface.obj : error LNK2001: unresolved external symbol "private: static class google::protobuf::internal::La
zyString const sentencepiece::TrainerSpec::_i_give_permission_to_break_this_code_default_eos_piece_" (?_i_give_permissi
on_to_break_this_code_default_eos_piece_@TrainerSpec@sentencepiece@@0VLazyString@internal@protobuf@google@@B) [C:\Users
\leres\Documents\GitHub\sentencepiece\src\sentencepiece_train.vcxproj]
sentencepiece_trainer.obj : error LNK2001: unresolved external symbol "private: static class google::protobuf::internal
::LazyString const sentencepiece::TrainerSpec::_i_give_permission_to_break_this_code_default_eos_piece_" (?_i_give_perm
ission_to_break_this_code_default_eos_piece_@TrainerSpec@sentencepiece@@0VLazyString@internal@protobuf@google@@B) [C:\U
sers\leres\Documents\GitHub\sentencepiece\src\sentencepiece_train.vcxproj]
trainer_interface.obj : error LNK2001: unresolved external symbol "private: static class google::protobuf::internal::La
zyString const sentencepiece::TrainerSpec::_i_give_permission_to_break_this_code_default_pad_piece_" (?_i_give_permissi
on_to_break_this_code_default_pad_piece_@TrainerSpec@sentencepiece@@0VLazyString@internal@protobuf@google@@B) [C:\Users
\leres\Documents\GitHub\sentencepiece\src\sentencepiece_train.vcxproj]
sentencepiece_trainer.obj : error LNK2001: unresolved external symbol "private: static class google::protobuf::internal
::LazyString const sentencepiece::TrainerSpec::_i_give_permission_to_break_this_code_default_pad_piece_" (?_i_give_perm
ission_to_break_this_code_default_pad_piece_@TrainerSpec@sentencepiece@@0VLazyString@internal@protobuf@google@@B) [C:\U
sers\leres\Documents\GitHub\sentencepiece\src\sentencepiece_train.vcxproj]
trainer_interface.obj : error LNK2019: unresolved external symbol "const sentencepiece::ModelProto::`vftable'" (??_7Mod
elProto@sentencepiece@@6B@) referenced in function "public: __cdecl sentencepiece::ModelProto::ModelProto(void)" (??0Mo
delProto@sentencepiece@@QEAA@XZ) [C:\Users\leres\Documents\GitHub\sentencepiece\src\sentencepiece_train.vcxproj]
unigram_model_trainer.obj : error LNK2001: unresolved external symbol "const sentencepiece::ModelProto::`vftable'" (??_
7ModelProto@sentencepiece@@6B@) [C:\Users\leres\Documents\GitHub\sentencepiece\src\sentencepiece_train.vcxproj]
sentencepiece_trainer.obj : error LNK2001: unresolved external symbol "const sentencepiece::ModelProto::`vftable'" (??_
7ModelProto@sentencepiece@@6B@) [C:\Users\leres\Documents\GitHub\sentencepiece\src\sentencepiece_train.vcxproj]
unigram_model_trainer.obj : error LNK2001: unresolved external symbol "class sentencepiece::TrainerSpecDefaultTypeInter
nal sentencepiece::_TrainerSpec_default_instance_" (?_TrainerSpec_default_instance_@sentencepiece@@3VTrainerSpecDefault
TypeInternal@1@A) [C:\Users\leres\Documents\GitHub\sentencepiece\src\sentencepiece_train.vcxproj]
sentencepiece_trainer.obj : error LNK2001: unresolved external symbol "private: static class google::protobuf::internal
::LazyString const sentencepiece::TrainerSpec::_i_give_permission_to_break_this_code_default_unk_surface_" (?_i_give_pe
rmission_to_break_this_code_default_unk_surface_@TrainerSpec@sentencepiece@@0VLazyString@internal@protobuf@google@@B) [
C:\Users\leres\Documents\GitHub\sentencepiece\src\sentencepiece_train.vcxproj]
sentencepiece_trainer.obj : error LNK2019: unresolved external symbol "const sentencepiece::TrainerSpec::`vftable'" (??
_7TrainerSpec@sentencepiece@@6B@) referenced in function "public: __cdecl sentencepiece::TrainerSpec::TrainerSpec(void)
" (??0TrainerSpec@sentencepiece@@QEAA@XZ) [C:\Users\leres\Documents\GitHub\sentencepiece\src\sentencepiece_train.vcxpro
j]
sentencepiece_trainer.obj : error LNK2019: unresolved external symbol "const sentencepiece::NormalizerSpec::`vftable'"
(??_7NormalizerSpec@sentencepiece@@6B@) referenced in function "public: __cdecl sentencepiece::NormalizerSpec::Normaliz
erSpec(class sentencepiece::NormalizerSpec &&)" (??0NormalizerSpec@sentencepiece@@QEAA@$$QEAV01@@Z) [C:\Users\leres\Doc
uments\GitHub\sentencepiece\src\sentencepiece_train.vcxproj]
C:\Users\leres\Documents\GitHub\sentencepiece\src\Debug\sentencepiece_train.dll : fatal error LNK1120: 9 unresolved ext
ernals [C:\Users\leres\Documents\GitHub\sentencepiece\src\sentencepiece_train.vcxproj]

```
Linkage error with  `spm_decode`, `spm_export_vocab` executable

```

spm_decode_main.obj : error LNK2001: unresolved external symbol "class absl::Flag<int> FLAGS_minloglevel" (?FLAGS_minlo
glevel@@3V?$Flag@H@absl@@A) [C:\Users\leres\Documents\GitHub\sentencepiece\src\spm_decode.vcxproj]
spm_decode_main.obj : error LNK2019: unresolved external symbol "const sentencepiece::SentencePieceText::`vftable'" (??
_7SentencePieceText@sentencepiece@@6B@) referenced in function "public: __cdecl sentencepiece::SentencePieceText::Sente
ncePieceText(void)" (??0SentencePieceText@sentencepiece@@QEAA@XZ) [C:\Users\leres\Documents\GitHub\sentencepiece\src\sp
m_decode.vcxproj]
C:\Users\leres\Documents\GitHub\sentencepiece\src\Debug\spm_decode.exe : fatal error LNK1120: 2 unresolved externals [C
:\Users\leres\Documents\GitHub\sentencepiece\src\spm_decode.vcxproj]
  Building Custom Rule C:/Users/leres/Documents/GitHub/sentencepiece/src/CMakeLists.txt
  spm_encode_main.cc
spm_encode_main.obj : error LNK2001: unresolved external symbol "class absl::Flag<int> FLAGS_minloglevel" (?FLAGS_minlo
glevel@@3V?$Flag@H@absl@@A) [C:\Users\leres\Documents\GitHub\sentencepiece\src\spm_encode.vcxproj]
spm_encode_main.obj : error LNK2019: unresolved external symbol "const sentencepiece::SentencePieceText::`vftable'" (??
_7SentencePieceText@sentencepiece@@6B@) referenced in function "public: __cdecl sentencepiece::SentencePieceText::Sente
ncePieceText(void)" (??0SentencePieceText@sentencepiece@@QEAA@XZ) [C:\Users\leres\Documents\GitHub\sentencepiece\src\sp
m_encode.vcxproj]
spm_encode_main.obj : error LNK2019: unresolved external symbol "const sentencepiece::NBestSentencePieceText::`vftable'
" (??_7NBestSentencePieceText@sentencepiece@@6B@) referenced in function "public: __cdecl sentencepiece::NBestSentenceP
ieceText::NBestSentencePieceText(void)" (??0NBestSentencePieceText@sentencepiece@@QEAA@XZ) [C:\Users\leres\Documents\Gi
tHub\sentencepiece\src\spm_encode.vcxproj]
C:\Users\leres\Documents\GitHub\sentencepiece\src\Debug\spm_encode.exe : fatal error LNK1120: 3 unresolved externals [C
:\Users\leres\Documents\GitHub\sentencepiece\src\spm_encode.vcxproj]
  Building Custom Rule C:/Users/leres/Documents/GitHub/sentencepiece/src/CMakeLists.txt
  spm_export_vocab_main.cc
spm_export_vocab_main.obj : error LNK2001: unresolved external symbol "class absl::Flag<int> FLAGS_minloglevel" (?FLAGS
_minloglevel@@3V?$Flag@H@absl@@A) [C:\Users\leres\Documents\GitHub\sentencepiece\src\spm_export_vocab.vcxproj]
C:\Users\leres\Documents\GitHub\sentencepiece\src\Debug\spm_export_vocab.exe : fatal error LNK1120: 1 unresolved extern
als [C:\Users\leres\Documents\GitHub\sentencepiece\src\spm_export_vocab.vcxproj]
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,6 +28,10 @@ option(SPM_NO_THREADLOCAL "Disable thread_local operator" OFF)
 option(SPM_USE_BUILTIN_PROTOBUF "Use built-in protobuf" ON)
 option(SPM_USE_EXTERNAL_ABSL "Use external abseil" OFF)
 
+# Disable shared build on windows
+if(WIN32)
+  set(SPM_ENABLE_SHARED OFF)
+
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 

